nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -131,9 +135,19 @@
     @WritesAttribute(attribute = "invokehttp.java.exception.message", description = "The Java exception message raised when the processor fails"),
     @WritesAttribute(attribute = "user-defined", description = "If the 'Put Response Body In Attribute' property is set then whatever it is set to "
         + "will become the attribute key and the value would be the body of the HTTP response.")})
-@DynamicProperty(name = "Header Name", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
-                    description = "Send request header with a key matching the Dynamic Property Key and a value created by evaluating "
-                            + "the Attribute Expression Language set in the value of the Dynamic Property.")
+@DynamicProperties ({
+    @DynamicProperty(name = "Header Name", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
+        description =
+            "Send request header with a key matching the Dynamic Property Key and a value created by evaluating "
+                + "the Attribute Expression Language set in the value of the Dynamic Property."),
+    @DynamicProperty(name = "post.form.<NAME>", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
+        description =
+            "When the HTTP Method is POST, dynamic properties with the property name in the form of post.form.<NAME>,"
+                + " where the <NAME> will be the form data name, will be used to fill out the multipart form parts."
+                + "  If the value is the literal 'FLOWFILE_CONTENT', that form part will have the flowfile's data."
+                + "  The property name post.form.filename will set the filename to set for the flowfile content."
```

Comment:
It's a bit of bad form to have a user-defined property with a well-known name. If the name is well-known it should not be user-defined. To be honest, I wouldn't even make this a configurable option. If the user wants a different filename sent, they should change the filename attribute of the FlowFile. This is how it's handled in almost all processors that deal with filenames. E.g., PutFile, PutHDFS, etc.

Reply:
ok, so you are saying to remove that, and have it set to the flow file's file name.  And document it?  and if they _don't_ want to set a file name, what do we do?  Have a post file name with form data properties ( boolean)?

Reply:
Correct. I would add a property that indicates whether or not to send the filename (probably would default to true??)

Reply:
What I did was add two properties:  the content name ( the form data name ) and a boolean to set the file name.   If that is set, we use the flow file file name.